# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmFrameIter.cpp
# Commit: a41351c7413b
# Full Hash: a41351c7413bfac7319da4af8d7437d620f71da8
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2022-03-23 04:09:13
# Regressor Bug: 1642412
# File Overlap Count: 7
# Description:
#   Bug 1642412 - wasm: Implement a 'precise' post-write barrier in baseline. r=jseward
#   
#   Add a post-write barrier which will remove a previous store buffer entry if the
#   new value stored in the field would not require it. This requires the previous
#   value to be loaded before the store and propagated to the post-write barrier
# ==============================================================================

diff -r d190ffcf25bf -r a41351c7413b js/src/wasm/WasmFrameIter.cpp
--- a/js/src/wasm/WasmFrameIter.cpp	Tue Mar 22 19:59:06 2022 +0000
+++ b/js/src/wasm/WasmFrameIter.cpp	Tue Mar 22 19:59:06 2022 +0000
@@ -1591,9 +1591,9 @@
     case SymbolicAddress::PreBarrierFiltering:
       return "call to native filtering GC prebarrier (in wasm)";
     case SymbolicAddress::PostBarrier:
-      return "call to native GC postbarrier (in wasm)";
+    case SymbolicAddress::PostBarrierPrecise:
     case SymbolicAddress::PostBarrierFiltering:
-      return "call to native filtering GC postbarrier (in wasm)";
+      return "call to native GC postbarrier (in wasm)";
     case SymbolicAddress::StructNew:
       return "call to native struct.new (in wasm)";
 #if defined(ENABLE_WASM_EXCEPTIONS)